kratos/utilities/particles_utilities.h

```diff
@@ -276,26 +274,48 @@ class ParticlesUtilities
                         (interpolations.second)[i] += shape_functions[k]*r_geom[k].GetValue(rInterpolationVariable);
 
                 }
-
             }
-
         });
 
         return interpolations;
-
     }
 
+    /**
+     * @brief This function takes a matrix of coordinates and gives the intrpolated value of the variable rInterpolationVariable
+    at those positions. Note that the value is only initialized if the particle is found
+     * @param rLocator this is a search structure for the volume
+     * @param rCoordinates positions at which we want to retrieve the interpolation values
+     * @param rInterpolationVariable: variable whose value that will be interpolated
+     * @param SearchTolerance: search tolerance used in the spatial search
+     * @return the function returns:
+           - is_inside a vector of bools telling if the given coordinate falls within the volume
+           - values the interpolated values (only valid if the corresponding value of is_inside is true)
+    */
+    template<unsigned int TDim, class TDataType, bool InterpolationVariableHasHistory>
+    static std::pair<std::vector<bool>, std::vector<TDataType> > InterpolateValuesAtCoordinates(
```

Comment:
here you are using again std::vector bool

Reply:
It is because pybind, doesn't work if not using std::vector or any compatible class. In all cases copy is necessary